File: kinotic-github/src/main/java/org/kinotic/github/api/model/ProjectGitHubRepoLink.java
package org.kinotic.github.api.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.kinotic.os.api.model.OrganizationScoped;

import java.util.Date;

/**
 * Persisted link between a Kinotic Project and a single GitHub repository reachable
 * through an existing {@link GitHubAppInstallation}. Drives webhook dispatch (delivery
 * → project) and ref-creation auth (project → which repo, via which installation).
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class ProjectGitHubRepoLink implements OrganizationScoped<String> {

    /** Stable Kinotic id; defaults to the projectId since each project has at most one link. */
    private String id;

    private String projectId;

    private String organizationId;

    /** Kinotic id of the {@link GitHubAppInstallation} this link routes through. */
    private String installationId;

    /** {@code owner/repo} string from GitHub. */
    private String repoFullName;

    /** GitHub's repo id (long, stored as string). Stable across renames. */
    private String repoId;

    private String defaultBranch;

    private Date updated;
}

File: kinotic-js/workspace/packages/os-api/src/api/model/github/ProjectGitHubRepoLink.ts
import type { Identifiable } from '@kinotic-ai/core'

/**
 * Persisted link between a Kinotic Project and a single GitHub repository reachable
 * through an existing {@link GitHubAppInstallation}. Drives webhook dispatch
 * (delivery → project) and ref-creation auth (project → which repo, via which
 * installation).
 */
export class ProjectGitHubRepoLink implements Identifiable<string> {
    public id: string | null = null
    public projectId: string = ''
    public organizationId: string = ''
    public installationId: string = ''
    public repoFullName: string = ''
    public repoId: number = 0
    public defaultBranch: string = ''
    public updated: number | null = null
}
